src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -151,12 +149,10 @@ internal sealed partial class ManagedWebSocket : WebSocket
         /// </summary>
         private object ReceiveAsyncLock => _utf8TextState; // some object, as we're simply lock'ing on it
 
-        /// <summary>Initializes the websocket.</summary>
-        /// <param name="stream">The connected Stream.</param>
-        /// <param name="isServer">true if this is the server-side of the connection; false if this is the client-side of the connection.</param>
-        /// <param name="subprotocol">The agreed upon subprotocol for the connection.</param>
-        /// <param name="keepAliveInterval">The interval to use for keep-alive pings.</param>
-        private ManagedWebSocket(Stream stream, bool isServer, string? subprotocol, TimeSpan keepAliveInterval)
+        private readonly WebSocketInflater? _inflater;
+        private readonly WebSocketDeflater? _deflater;
+
+        private ManagedWebSocket(Stream stream, WebSocketCreationOptions options)
```

Comment:
This is gonna result in more allocations per websocket since we unify to the `WebSocketCreationOptions `.

Reply:
Do you suggest I should extend the original constructor by including `WebSocketDeflateOptions` as parameter, instead?

Reply:
Or maybe have 2 constructors.